packages/remix-dev/compiler/manifest.ts

```diff
@@ -107,7 +107,7 @@ export const write = async (config: RemixConfig, assetsManifest: Manifest) => {
 
   await writeFileSafe(
     path.join(config.assetsBuildDirectory, filename),
-    `window.__remixManifest=${JSON.stringify(assetsManifest)};`
+    `window.__remixManifest=JSON.parse('${JSON.stringify(assetsManifest)}');`
```

Comment:
Wrapping this in single quote's isn't guaranteed to work if the object contains any single quotes.  We should switch this over to a double `JSON.stringify` like we do in `react-router-dom` `StaticRouterProvider`.

<img width="580" alt="Screenshot 2023-05-02 at 9 27 30 AM" src="https://user-images.githubusercontent.com/1609022/235681616-4745bd77-1c54-484e-b7df-d821a07bceaa.png">

Reply:
Updated in [4739fbf](https://github.com/remix-run/remix/pull/5555/commits/4739fbf945415aa3520d24ddee34d4f732801a69)

Reply:
The only concern with the double `JSON.stringify` is the added bytes over the wire due to the `\\` escaping though - but the manifest oughta be cached to alleviate that after the first load?